src/Common/FailPoint.h

```diff
@@ -0,0 +1,47 @@
+#pragma once
+#include "config.h"
+
+#include <Common/Exception.h>
+#include <Core/Types.h>
+
+#ifdef __clang__
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdocumentation"
+#pragma clang diagnostic ignored "-Wreserved-macro-identifier"
+#endif
+
+#include <fiu.h>
+#include <fiu-control.h>
+
+#ifdef __clang__
+#pragma clang diagnostic pop
+#endif
+
+#include <any>
+#include <unordered_map>
+
+namespace DB
+{
+
+// When `fail_point` is enabled, wait till it is disabled
+#define FAIL_POINT_PAUSE(fail_point) fiu_do_on(fail_point, FailPointHelper::wait(fail_point);)
+
+class FailPointChannel;
+class FailPointHelper
+{
+public:
+    static void enableFailPoint(const String & fail_point_name, std::optional<std::any> v = std::nullopt);
+
+    static std::optional<std::any> getFailPointVal(const String & fail_point_name);
```

Comment:
Idea of failpoint values is foreign to the libfiu and is not used yet in our code. Do we have scenarios when we will use it? Maybe add at least a simple test for it?

It will be hard to use this feature in case of multiple failures (not "once" mode). So I'm inclined to remove it at all to avoid complications.